src/org/zaproxy/zap/extension/spider/SpiderThread.java

```diff
@@ -589,4 +630,8 @@ public void setCustomParseFilters(List<ParseFilter> customParseFilters) {
 		this.customParseFilters  = customParseFilters;
 	}
 
+    public int getNumberOfNodesAdded() {
+        return getAddedNodesTableModel().getRowCount();
```

Comment:
The table model is the only place we currently record this. Is it really worth having another variable which should just be the same as the row count? I dont think there should be any sync issues, but if they are they'll just be transitory wont they?